### PR TITLE
created settings per address in the broker

### PR DESCRIPTION
**Pull request by vbusch**
### Type of change

- Enhancement / new feature

### Description
issue #3157 
- in the standard address space and brokered these are updated when we an address plan is changed, or when an address is created / deleted / modified

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: agent/lib/broker_address_settings.js [hunk 1]
```diff
@@ -31,52 +43,100 @@ function extract_address_plan(object) {
...
-    return new Promise(function (resolve, reject) {
-        self.watcher.once('updated', function () {
-            log.info('plans have been retrieved');
+BrokerAddressSettings.prototype.get_changes = function (name, addressplans, unchanged) {
+    var c = myutils.changes(this.last[name], addressplans, addressplan_compare, unchanged, description);
+    this.last[name] = addresses;
+    return c;
+};
+
+BrokerAddressSettings.prototype.get_global_max_size = function () {
```
Comment (lulf) [inline agent/lib/broker_address_settings.js:97]: I think a simpler way to get this would be to inject the value as an environment variable in the template..

File: agent/lib/broker_address_settings.js [hunk 2]
```diff
@@ -18,9 +18,21 @@
 var log = require("./log.js").logger();
 var kubernetes = require('./kubernetes.js');
 var myutils = require('./utils.js');
+var events = require('events');
+var util = require('util');
 
 function kubernetes_resource_compare(a, b) {
-    return myutils.string_compare(a.metadata.name, b.metadata.name);
+    return myutils.string_compare(a.name, b.name);
```
Comment (k-wall) [inline agent/lib/broker_address_settings.js:25]: I think this is used to compare address settings, rather than Kubernetes resources, if so, it should be renamed.

File: agent/lib/broker_address_settings.js [hunk 3]
```diff
@@ -18,9 +18,21 @@
...
+}
+
+function extract_spec(def) {
```
Comment (k-wall) [inline agent/lib/broker_address_settings.js:28]: name does not reflect the implementation well.  How about something like flatten_plan?

File: agent/lib/broker_address_settings.js [hunk 4]
```diff
@@ -86,7 +146,7 @@ BrokerAddressSettings.prototype.delete_settings = function (plan) {
 };
 
 BrokerAddressSettings.prototype.updated = function (objects) {
-    var plans = objects.map(extract_address_plan).filter(required_broker_resource);
+    var plans = objects.filter((plan) => plan !== undefined).map(extract_spec).filter(required_broker_resource);
```
Comment (k-wall) [inline agent/lib/broker_address_settings.js:149]: is it really necessary to filter for undefined objects? sounds fishy to me.

File: agent/lib/broker_controller.js [hunk 5]
```diff
@@ -89,6 +100,28 @@ BrokerController.prototype.sync_addresses = function (addresses) {
...
+    addressplans.forEach(function (addressplan) {
+
+        for (var address in self.addresses) {
+            if (self.addresses[address].plan === addressplan.name) {
+                addrs.push(self.addresses[address]);
+            }
+        }
+    });
+        if (addrs.length) {
+            return Promise.all(addrs.map(function (addr) {
```
Comment (k-wall) [inline agent/lib/broker_controller.js:115]: Should we apply the 'p-limit' pattern here?

File: agent/lib/broker_controller.js [hunk 6]
```diff
@@ -89,6 +100,28 @@ BrokerController.prototype.sync_addresses = function (addresses) {
...
+                return self.update_address_setting(addr);
+            })).then(function () {log.info('updated all ')})
```
Comment (k-wall) [inline agent/lib/broker_controller.js:117]: make this logging more specific, or remove it.

File: agent/lib/broker_controller.js [hunk 7]
```diff
@@ -248,6 +281,17 @@ function is_not_internal(conn) {
     return conn.user !== undefined && conn.user.match('^agent.[a-z0-9]+$') == null; //Can't get properties or anything else on which to base decision yet
 }
 
+function compare_address_settings(orig_address_settings, new_address_settings) {
+    for (var name in orig_address_settings) {
+        log.debug('comparing %s: %s %s', name, orig_address_settings[name], new_address_settings[name])
+        if (new_address_settings[name] && myutils.string_compare(orig_address_settings[name], new_address_settings[name]) !=0){
```
Comment (k-wall) [inline agent/lib/broker_controller.js:287]: recompilation of string_compare in the false case.

File: agent/lib/broker_controller.js [hunk 8]
```diff
@@ -248,6 +281,17 @@ function is_not_internal(conn) {
     return conn.user !== undefined && conn.user.match('^agent.[a-z0-9]+$') == null; //Can't get properties or anything else on which to base decision yet
 }
 
+function compare_address_settings(orig_address_settings, new_address_settings) {
+    for (var name in orig_address_settings) {
```
Comment (k-wall) [inline agent/lib/broker_controller.js:285]: this method won't catch the case where the new settings object has a value that the old  does not.  Is this important?

File: agent/lib/broker_controller.js [hunk 9]
```diff
@@ -383,40 +427,58 @@ function translate(addresses_in, excluded_names, excluded_types) {
     return addresses_out;
 }
 
+BrokerController.prototype.update_address_setting = function (a) {
+    var self = this;
+    return self.broker.getAddressSettings(a.address).then(function (orig_settings) {
+
+        return self.get_address_settings(a).then(function (new_settings) {
+            if (compare_address_settings(orig_settings, new_settings) != 0) {
+                self.broker.removeAddressSettings(a.address).then(function () {
```
Comment (k-wall) [inline agent/lib/broker_controller.js:436]: error handling missing

File: agent/lib/broker_controller.js [hunk 10]
```diff
@@ -383,40 +427,58 @@ function translate(addresses_in, excluded_names, excluded_types) {
...
-        log.info('[%s] Deleting queue "%s"...', self.id, a.address);
-        return self.broker.destroyQueue(a.address).then(function () {
-            log.info('[%s] Deleted queue "%s"', self.id, a.address);
-            self.post_event(myevents.address_delete(a));
-        }).catch(function (error) {
-            log.error('[%s] Failed to delete queue %s: %s', self.id, a.address, error);
-            return self.broker.deleteAddress(a.address).then(function () {
-                log.info('[%s] Deleted anycast address %s', self.id, a.address);
+
+    return self.broker.getAddressSettings(a.address).then(function (settings) {
```
Comment (k-wall) [inline agent/lib/broker_controller.js:447]: why do we need to lookup the address settings here?  also, if the address settings are missing is it correct to not delete the queues etc?

File: agent/lib/broker_controller.js [hunk 11]
```diff
@@ -426,14 +488,17 @@ BrokerController.prototype.delete_address_and_settings = function (a) {
         log.info('[%s] Deleted address-settings for "%s"', self.id, a.address);
         return self.delete_address(a);
     }).catch(function (error) {
-        log.error('[%s] Failed to delete address settings for "%s": %s', self.id, a.address, error);
+        log.error('[%s] Failed to delete address setting for "%s": %s', self.id, a.address, error);
```
Comment (k-wall) [inline agent/lib/broker_controller.js:491]: I don't understand the need for the two catches.  Would moving the self.delete_address(a) to a .finally() make the code simpler?

File: agent/lib/broker_controller.js [hunk 12]
```diff
@@ -476,18 +541,50 @@ BrokerController.prototype.create_address = function (a) {
...
+                        })
+                        .catch(function (error) {
+                            log.error('[%s] Failed to create address settings for "%s": %s', self.id, a.address, error);
+                            return self.create_address(a);
+                        });
+                });
+        });
 };
 
+BrokerController.prototype.ensure_connector = function (a) {
```
Comment (k-wall) [inline agent/lib/broker_controller.js:561]: Can you tell me more about this part of the change?

File: agent/lib/broker_address_settings.js [hunk 13]
```diff
@@ -31,52 +43,100 @@ function extract_address_plan(object) {
     }
 }
 
+function addressplan_compare(a, b) {
+    return myutils.string_compare(a.name, b.name);
+}
+
```
Comment (k-wall) [inline agent/lib/broker_address_settings.js:49]: let's name this function so the name doesn't collide with the map.

**Comment by k-wall** [comment]
@lulf @vbusch I don't understand how this feature is supposed to work with PAGE policy.  If I infer the intent properly from the code we allocate a plan defined proportion of the globalMaxSize to each address, so in the PAGE case, the address' size would be constrained by RAM.

File: agent/lib/broker_address_settings.js [hunk 14]
```diff
@@ -31,52 +43,100 @@ function extract_address_plan(object) {
...
-        var r = this.required_broker_resource[plan];
-        if (r) {
-            return {
-                maxSizeBytes: r * global_max_size,
-                addressFullMessagePolicy: 'FAIL'
-            };
-        } else {
-            log.info('no broker resource required for %s, therefore not applying address settings', plan);
-        }
+    var r = this.required_broker_resource[plan];
```
Comment (k-wall) [inline agent/lib/broker_address_settings.js:132]: Doesn't this calculation need to factor in partitions?

File: agent/lib/broker_address_settings.js [hunk 15]
```diff
@@ -31,52 +43,100 @@ function extract_address_plan(object) {
...
+    if (r) {
+        return {
+            maxSizeBytes: Math.round(r * global_max_size),
+            addressFullMessagePolicy: 'FAIL'
+        };
     } else {
-        log.info('no global max, therefore not applying address settings');
+        log.info('no broker resource required for %s, therefore not applying address settings', plan);
     }
 };
```
Comment (k-wall) [inline agent/lib/broker_address_settings.js:142]: in delete_settings, I think you are passing the result of extract_spec, so plan.metadata will be undefined and will crash.

File: agent/lib/broker_address_settings.js (hunk 2 above)
Comment (vbusch) [reply agent/lib/broker_address_settings.js]: Renamed to 'addressplan_name_compare'

File: agent/lib/broker_address_settings.js (hunk 4 above)
Comment (vbusch) [reply agent/lib/broker_address_settings.js]: was able to remove the filter

File: agent/lib/broker_controller.js (hunk 5 above)
Comment (vbusch) [reply agent/lib/broker_controller.js]: added p-limit

File: agent/lib/broker_controller.js (hunk 6 above)
Comment (vbusch) [reply agent/lib/broker_controller.js]: removed

File: agent/lib/broker_controller.js (hunk 7 above)
Comment (vbusch) [reply agent/lib/broker_controller.js]: updated

File: agent/lib/broker_controller.js (hunk 8 above)
Comment (vbusch) [reply agent/lib/broker_controller.js:281]: The broker returns all the properties with default values.  It is important that we only compare the ones in the new_settings.

File: agent/lib/broker_controller.js (hunk 9 above)
Comment (vbusch) [reply agent/lib/broker_controller.js]: added

File: agent/lib/broker_controller.js (hunk 11 above)
Comment (vbusch) [reply agent/lib/broker_controller.js:503]: removed the extra

File: agent/lib/broker_controller.js (hunk 12 above)
Comment (vbusch) [reply agent/lib/broker_controller.js]: wasn't required

**Comment by vbusch** [comment]
@enmasse-ci run tests profile=isolated

File: agent/lib/artemis.js [hunk 16]
```diff
@@ -499,7 +499,7 @@ var address_settings_fields = {
     'addressFullMessagePolicy': 'FAIL',
     'slowConsumerThreshold':-1,
     'slowConsumerCheckPeriod':-1,
-    'slowConsumerPolicy':'DROP',
+    'slowConsumerPolicy':'KILL',
```
Comment (lulf) [inline agent/lib/artemis.js:502]: Why this change?

File: agent/lib/broker_address_settings.js [hunk 17]
```diff
@@ -18,65 +18,90 @@
 var log = require("./log.js").logger();
 var kubernetes = require('./kubernetes.js');
 var myutils = require('./utils.js');
+var events = require('events');
+var util = require('util');
 
-function kubernetes_resource_compare(a, b) {
-    return myutils.string_compare(a.metadata.name, b.metadata.name);
+function extract_flatten_plan(def) {
```
Comment (lulf) [inline agent/lib/broker_address_settings.js:24]: Is there any real need to 'flatten' the definition? Iirc the reason this is done for addresses is because of legacy configuration format from looong ago.

File: agent/lib/utils.js [hunk 18]
```diff
@@ -286,3 +286,26 @@ module.exports.get = function (object, fields, default_value) {
...
+var multipliers = {};
+multipliers.B = 1;
+multipliers.KB = ( multipliers.B * 1024 );
+multipliers.MB = ( multipliers.KB * 1024 );
+multipliers.GB = ( multipliers.MB * 1024 );
+module.exports.parseToBytes = function (input) {
+    var unit = input.toUpperCase().match(/([KMGTPEZ]I?)?B$/);
+
+    if (!unit) {
+        log.warn( "parseToBytes input does not contain a supported unit of measurement.  Using 'B': %s", input );
```
Comment (lulf) [inline agent/lib/utils.js:299]: We should probably use 'k' or both 'k' and 'K', because 'k' is the official.

File: agent/lib/artemis.js (hunk 16 above)
Comment (vbusch) [reply agent/lib/artemis.js:502]: DROP has been removed.  The available options are kill & notify
https://github.com/apache/activemq-artemis/blob/master/artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/SlowConsumerPolicy.java

File: agent/lib/broker_address_settings.js (hunk 17 above)
Comment (vbusch) [reply agent/lib/broker_address_settings.js:24]: ah, I'll change that.  I saw it was done elsewhere and followed.

File: agent/lib/utils.js (hunk 18 above)
Comment (vbusch) [reply agent/lib/utils.js:299]: by switching it to upper case(toUpperCase) both will be allowed.

Comment (lulf) [reply agent/lib/utils.js:299]: Ah, i didn't catch that

File: agent/lib/artemis.js [hunk 19]
```diff
@@ -523,6 +523,22 @@ Artemis.prototype.removeAddressSettings = function (match) {
...
+Artemis.prototype.getAddressSettings = function (match) {
+    return this._request('broker', 'getAddressSettingsAsJSON', [match]).then(function (result) {
+        return JSON.parse(result);
+    });
+};
+
+Artemis.prototype.listAddressSettings = function () {
+    var self = this;
+    return this.listAddresses().then(function (addresses) {
+        return Promise.all(addresses.map(function (a) {
```
Comment (k-wall) [inline agent/lib/artemis.js:535]: suggest using the `plimit` pattern to reduce the inflight.

Comment (vbusch) [reply agent/lib/artemis.js:535]: I'll remove that method.  broker_controller already has the list of addresses, and is using that list to call getAddressSettings (with p-limit).  
This one was the original, that is no longer used.

File: agent/lib/broker_address_settings.js (hunk 14 above)
Comment (k-wall) [reply agent/lib/broker_address_settings.js:97]: @lulf @vbusch if you have a plan that requires 0.5 broker and you are partitioning across 2, I think that means you should have 0.25 per broker.  I therefore think the calculation needs to divide by patitions.  Is that right?

File: agent/lib/broker_address_settings.js [hunk 20]
```diff
@@ -86,34 +111,35 @@ BrokerAddressSettings.prototype.delete_settings = function (plan) {
...
-        return this.wait_for_plans().then(function () {
-            return global_max_size_promise.then(function (global_max_size) {
-                var settings = self.generate_address_settings(address.plan, global_max_size);
-                log.info('using settings %j for %s', settings, address.address);
-                return settings;
-            });
-        });
-    } else {
+
+    var max_size_as_bytes = this.config.BROKER_GLOBAL_MAX_SIZE ? myutils.parseToBytes(this.config.BROKER_GLOBAL_MAX_SIZE) : 0;
```
Comment (k-wall) [inline agent/lib/broker_address_settings.js:132]: Would it be cleaner to read GLOBAL_MAX_SIZE/request max size from the Broker once in `start()`  After all the value is not going to change.  

something like:

```
BrokerAddressSettings.prototype.start = function () {
...
read_global_max_size().then((size) => {
self.global_max_szie = size;
self.watcher.on('updated', self.updated.bind(this));
});
```

File: agent/lib/broker_controller.js [hunk 21]
```diff
@@ -383,8 +427,38 @@ function translate(addresses_in, excluded_names, excluded_types) {
...
+
+                    return self.broker.removeAddressSettings(a.address).then(function () {
+
+                        return self.broker.addAddressSettings(a.address, new_settings);
+                    }).catch(function (error) {
+                        log.error('[%s] Adding the new address setting (removal failed )%s: %s', self.id, a.address, error);
+                        return self.broker.addAddressSettings(a.address, new_settings)
+                    });
+                } else {
+                    log.info('[%s] Address settings match for %s: not updating', self.id, a.address);
```
Comment (k-wall) [inline agent/lib/broker_controller.js:446]: To verbose (systems that have many addresses). Change to debug.

File: agent/lib/utils.js [hunk 22]
```diff
@@ -286,3 +286,26 @@ module.exports.get = function (object, fields, default_value) {
     }
     return o || default_value;
 };
+
+var multipliers = {};
+multipliers.B = 1;
+multipliers.KB = ( multipliers.B * 1024 );
+multipliers.MB = ( multipliers.KB * 1024 );
+multipliers.GB = ( multipliers.MB * 1024 );
+module.exports.parseToBytes = function (input) {
```
Comment (k-wall) [inline agent/lib/utils.js:295]: it seems weird that the regex allows GTPEZ but the multipler map does not.

File: agent/lib/broker_controller.js [hunk 23]
```diff
@@ -383,8 +427,38 @@ function translate(addresses_in, excluded_names, excluded_types) {
...
+                }
+            } else {
+                //Settings weren't created, and reason is already logged. Most likely broker resource not required.
+            }
+        }).catch(function (error) {
+            log.error('[%s] Failed to create new address setting %s: %s', self.id, a.address, error);
```
Comment (k-wall) [inline agent/lib/broker_controller.js:452]:
```suggestion
            log.error('[%s] Failed to create new address setting %s: %s', self.id, a.address, error);
            return Promise.reject(error);
```

File: agent/lib/broker_controller.js [hunk 24]
```diff
@@ -383,8 +427,38 @@ function translate(addresses_in, excluded_names, excluded_types) {
...
+        });
+    }).catch(function (error) {
+        log.error('[%s] Failed to retrieve brokers address setting: %s', self.id, a.address, error);
```
Comment (k-wall) [inline agent/lib/broker_controller.js:455]:
```suggestion
        log.error('[%s] Failed to retrieve brokers address setting: %s', self.id, a.address, error);
        return Promise.reject(error);
```

File: agent/lib/broker_controller.js (hunk 21 above)
Comment (k-wall) [inline agent/lib/broker_controller.js:446]:
```suggestion
                    log.info('[%s] Address settings match for %s: not updating', self.id, a.address);
                    return Promise.resolve();
```

File: agent/lib/broker_controller.js [hunk 25]
```diff
@@ -383,8 +427,38 @@ function translate(addresses_in, excluded_names, excluded_types) {
...
+                        return self.broker.addAddressSettings(a.address, new_settings);
+                    }).catch(function (error) {
+                        log.error('[%s] Adding the new address setting (removal failed )%s: %s', self.id, a.address, error);
+                        return self.broker.addAddressSettings(a.address, new_settings)
+                    });
+                } else {
+                    log.info('[%s] Address settings match for %s: not updating', self.id, a.address);
+                }
+            } else {
+                //Settings weren't created, and reason is already logged. Most likely broker resource not required.
```
Comment (k-wall) [inline agent/lib/broker_controller.js:449]:
```suggestion
                //Settings weren't created, and reason is already logged. Most likely broker resource not required.
                return Promise.resolve();
```

File: agent/lib/artemis.js (hunk 19 above)
Comment (k-wall) [reply agent/lib/artemis.js]: ok

File: agent/lib/broker_address_settings.js (hunk 14 above)
Comment (k-wall) [reply agent/lib/broker_address_settings.js:87]: @lulf @vbusch is my comment invalid?

Comment (lulf) [reply agent/lib/broker_address_settings.js:87]: Sorry, I have forgotten to respond to this. Yes, the credits should be divided by the number of partitions.

**Comment by vbusch** [comment]
@enmasse-ci run tests profile=isolated

**Comment by vbusch** [comment]
@enmasse-ci run tests profile=isolated

**Comment by vbusch** [comment]
Investigated the failing testcase "isolated.CommonTest#testRestartComponents"
It passes on master.  But the logs on master show the same issue.
There are 2 address spaces, brokered & standard.  Each address space has the same queue test-queue.  When restarting all of enmasse (restarting single components works), the agent of the standard address space has an issue with configuring its router.
It is getting all the addresses (for both address spaces), then it tries to create the test-queue address twice.  And fails the 2nd time (ERROR creating address test-queue on qdrouterd-6c2f4cf-0 => Name conflicts with an existing entity)
It does retry and then gets (ERROR configuration update for qdrouterd-6c2f4cf-0 not up to date (attempt 3 of 3), agent ERROR Unable to apply desired configuration; gave up after 3 attempts)
I haven't determined yet, why the test passes on master.  But it doesn't seem right that the standard address space is creating addresses that belong in the brokered address space.

**Comment by k-wall** [comment]
@lulf @vbusch reflecting in this change, I'm not sure we have done the right thing.   I think we should be driving the address settings based on the `status.planStatus` section of each address and _not_ the address plan directly. Taking the it from planStatus is the correct thing to do because it means that the address's controller has had chance to apply the validations.  Also it has an implementation benefit - the additional events that @vbusch has had to introduce to deal with plan updates make for complicated and complex code that is hard to reason about.  Making this change would get rid of this complication.

For standard - the status.planStatus is already in place.  For brokered, this needs an extension to the controller (agent), but this is code I most have already on the TTL change.

**Comment by vbusch** [comment]
@k-wall This morning, I want to quickly confirm whether or not we need to delete the per address settings, before re-adding them for the update.  I saw some behaviour last night, that suggests the delete this might be problematic.  But could be simply solved, if an add just overwrites.
  Driving it from the `status.planStatus` is an interesting idea.  I'll have a look at that for the standard.  If it simplifies the code, that would be great.

**Comment by k-wall** [comment]
@vbusch see my comment on #4612.  I think with the way you chain the creation of address settings before addresses, that route is more like to create the address.

**Comment by vbusch** [comment]
Pushed the change with the system tests + refactoring to use the addresses.status.planStatus.
I broke the update on my final cleanup.  I believe it might be a change I made to BrokerController.prototype.on_connection_open or the addresses structure.  Will look at it first thing in the am.

**Comment by vbusch** [comment]
@enmasse-ci run tests profile=isolated

File: systemtests/src/test/java/io/enmasse/systemtest/bases/plans/PlansTestBase.java [hunk 26]
```diff
@@ -131,6 +144,65 @@ public void doTestUnknownAddressPlan(AddressSpace addressSpace, List<StageHolder
         } while(true);
     }
 
+    public void doTestUpdatePlanBrokerCredit_changesPerAddressMaxSize(AddressSpace addressSpace, Address queueDest, AddressPlan phase1, AddressPlan phase2, AddressPlan redefinedPhase2, AmqpClientFactory amqpClientFactory) throws Exception {
```
Comment (kornys) [inline systemtests/src/test/java/io/enmasse/systemtest/bases/plans/PlansTestBase.java:147]: Please use java camel syntax instead of `_`

File: systemtests/src/test/java/io/enmasse/systemtest/isolated/plans/brokered/PlansTestBrokered.java [hunk 27]
```diff
@@ -149,4 +152,75 @@ void testUnknownAddressPlan() throws Exception {
         doTestUnknownAddressPlan(addressSpace, stageHolders);
     }
 
+    @Test
+    void testUpdatePlanBrokerCredit_changesPerAddressMaxSize() throws Exception {
```
Comment (kornys) [inline systemtests/src/test/java/io/enmasse/systemtest/isolated/plans/brokered/PlansTestBrokered.java:156]: same here please avoid using `_`

File: systemtests/src/test/java/io/enmasse/systemtest/isolated/plans/standard/PlansTestStandard.java [hunk 28]
```diff
@@ -1168,6 +1168,88 @@ void testUnknownAddressPlan() throws Exception {
         doTestUnknownAddressPlan(addressSpace, stageHolders);
     }
 
+    @Test
+    void testUpdatePlanBrokerCredit_changesPerAddressMaxSize() throws Exception {
```
Comment (kornys) [inline systemtests/src/test/java/io/enmasse/systemtest/isolated/plans/standard/PlansTestStandard.java:1172]: same here please avoid using `_` and also add this test into acceptance profile then we can see behavior on ocp4

**Comment by vbusch** [comment]
Some initial times for creating queues
```
				PR	Master
create 2500 queue:		7m30s	6m59s
create additional 1:		55s	58s
create additional 2500:		15:30	5:52+(missed recording the time where they were all in ready state)	
create additional 1:		2m15s	2m32s
delete all queues		14m56s	20m2s	

```
Recorded on a 3.11 cluster

**Comment by vbusch** [comment]
@enmasse-ci run tests profile=isolated

File: systemtests/src/test/java/io/enmasse/systemtest/bases/plans/PlansTestBase.java (hunk 26 above)
Comment (k-wall) [reply systemtests/src/test/java/io/enmasse/systemtest/bases/plans/PlansTestBase.java]: Fixed by @vbusch
